en/ros2/px4_sdk.md

```diff
@@ -0,0 +1,336 @@
+# PX4 SDK
+
+The [PX4 SDK](https://github.com/PX4/px4_sdk) is a C++ library to simplify controlling PX4 from ROS 2.
+It allows to write external modes that are dynamically registered with PX4 and behave the same way as internal ones.
+A mode can send different types of setpoints, ranging from high-level navigation tasks all the way down to direct actuator controls.
+
+## Overview
+
+This diagram provides a conceptual overview:
+
+![ROS2 modes overview diagram](../../assets/middleware/ros2/px4_sdk/ros2_modes_overview.svg)
+
+<!-- Source: https://docs.google.com/drawings/d/1WByCfgcytnaow7r41VhYJL8OGrw1RjFO51GoPMQBCNA/edit -->
+
+The following sections define and explain the terms used in the diagram.
+
+### Definitions
+
+#### Mode
+
+- A mode is a component that can send setpoints to the vehicle, which control its motion (such as velocity or direct actuator commands).
+- It is passive, meaning it does not activate other modes (by sending commands), and it must be activated by someone else (either the user through RC/GCS, the FMU in a failsafe situation or by a _mode executor_).
+- It selects a setpoint type and sends it while it is active.
+  A mode can switch between multiple setpoint types.
+- Has a name displayed by the GCS.
+- Can configure its mode requirements (for example that it requires a valid position estimate).
+- A mode can consist of different tasks (for example flying to a target, lowering a winch, releasing a payload and then fly back).
+
+#### Mode Executor
+
```

Comment:
It would be good to pull out one of these as an introductory sentence. I move the topic "When would you use a mode executor?" - something like this:

```suggestion
A mode executor is an optional component, associated with a mode, for putting the vehicle into a state where the mode can be activated.
For example, the mode executor for a payload delivery mode might first trigger a take-off, then switch to the mode to deliver the payload, and when that completes trigger an RTL.
```

FMI is the state machine flexible? I.e. my executor for above case might be called when already in the air - can it skip the takeoff state?